Mobile/ShopLens/Droid/Activities/MainActivity.cs

```diff
@@ -103,59 +115,145 @@ protected override void OnCreate(Bundle savedInstanceState)
-                        StartActivity(intentSpeech);
-                        break;
                 }
             };
         }
 
-        void RecogniseVoice(object sender, EventArgs e)
+        protected override void OnStart()
+        {
+            base.OnStart();
+            tutorialNeeded = IsTutorialNeeded();
+        }
+
+        protected override void OnStop()
         {
-            if (commandRecognizer.IsValueCreated)
+            // Get rid of TTS.
+            if (tts != null)
             {
-                speechIntent = VoiceRecognizerHelper.SetUpVoiceRecognizerIntent();
-                commandRecognizer.Value.StartListening(speechIntent);
+                tts.Stop();
+                tts.Shutdown();
             }
+            base.OnStop();
+        }
+
+        private bool IsTalkBackEnabled()
+        {
+            var accessManager = (AccessibilityManager)GetSystemService(AccessibilityService);
+            return accessManager.IsTouchExplorationEnabled;
+        }
+
+        private void InitiateNoTalkBackMode()
+        {
+            needUserAnswerId = ConfigurationManager.AppSettings["AnswerUtteranceId"];
+            askUserToRepeat = ConfigurationManager.AppSettings["AskToRepeat"];
+
+            ttsListener = new ShopLensUtteranceProgressListener(TtsStoppedSpeaking);
+
+            tts = new TextToSpeech(this, this);
+            tts.SetOnUtteranceProgressListener(ttsListener);
+
+            voiceRecognizer = new ShopLensSpeechRecognizer(OnVoiceRecognitionResults);
+
+            noTalkBack = true;
         }
 
-        // When the current voice recognition session stops.
-        public void OnResults(Bundle results)
+        private bool IsTutorialNeeded()
         {
-            var matches = results.GetStringArrayList(SpeechRecognizer.ResultsRecognition);
-            if (matches.Count > 0)
+            bool firstTime = prefs.GetBoolean("FirstTime", true);
+
+            if (firstTime)
             {
-                if (matches[0] == ConfigurationManager.AppSettings["CmdOpenCamera"])
-                {
-                    var intent = new Intent(this, typeof(CameraActivity));
-                    StartActivity(intent);
-                }
-                else if (matches[0] == ConfigurationManager.AppSettings["CmdOpenCart"])
-                {
-                    var intent = new Intent(this, typeof(ShoppingCartActivity));
-                    StartActivity(intent);
-                }
-                else if (matches[0] == ConfigurationManager.AppSettings["CmdOpenList"])
-                {
-                    var intent = new Intent(this, typeof(ShoppingListActivity));
-                    StartActivity(intent);
-                }
-                // For debugging purposes.
-                else
-                {
-                    voiceCommandButton.Text = matches[0];
-                }
+                prefs.Edit().PutBoolean("FirstTime", false).Commit();
```

Comment:
Not the best solution for detecting whether the app has been launched on the device once or not, because preferences get reset with every new build deployed to the device, but for now I think it's good.